lib/services/keyVault/lib/keyvault.d.ts

```diff
@@ -42,7 +42,7 @@ export interface Authenticator {
  * @param {KeyVaultCredentials~authRequest} authenticator  A callback that receives a challenge and returns an authentication token.
  */
 export class KeyVaultCredentials implements msRest.ServiceClientCredentials {
-  cosntructor(authenticator: Authenticator);
+  constructor( authenticator:  (challenge: any, callback: any) => any );
```

Comment:
it turns out this was about the best signature I could come up with as well. At the maximum, I was able to strongly type `challenge`, which makes it no more useful than this. Also,  this `authenticator` function needs `adal-node` library, which itself is untyped at the moment. For these reasons, I agree that `(any, any) => any` is the best we could do now. We can revisit this in the future, if needed.